### PR TITLE
DPR-198: Process CDC events and archive to raw zone

### PR DESCRIPTION
**Pull request by koladeadewuyi-moj**
Changes included in this PR:
- Process CDC events [DPR-309](https://dsdmoj.atlassian.net/browse/DPR-309)
- Archive CDC events to the raw zone [DPR-310](https://dsdmoj.atlassian.net/browse/DPR-310)

[DPR-309]: https://dsdmoj.atlassian.net/browse/DPR-309?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DPR-310]: https://dsdmoj.atlassian.net/browse/DPR-310?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/converter/dms/DMS_3_4_6.java [hunk 1]
```diff
@@ -81,6 +82,8 @@ public static Optional<Operation> getOperation(String operation) {
         }
     }
 
+    public static DMS_3_4_6.Operation[] cdcOperations = { Insert, Update, Delete };
```
Comment (carwynmoj) [inline src/main/java/uk/gov/justice/digital/converter/dms/DMS_3_4_6.java:85]: I think it's better to have this as a property on the `Operation` enum. See my comment re. the set further down.

File: src/main/java/uk/gov/justice/digital/zone/RawZone.java [hunk 2]
```diff
@@ -30,6 +34,11 @@ public class RawZone extends Zone {
     private final DataStorageService storage;
     private final SourceReferenceService sourceReferenceService;
 
+    private final Set<String> cdcOperationsSet = Arrays
```
Comment (carwynmoj) [inline src/main/java/uk/gov/justice/digital/zone/RawZone.java:37]: I think it'd be good to have this on the `Operation` enum. So we could have something like `getCdcOperations` that returns a set. 

The reasoning being that this way, we only need to define this information once, and the definition is located right next to the definiton of the values themselves.

File: src/main/java/uk/gov/justice/digital/zone/RawZone.java [hunk 3]
```diff
@@ -40,58 +49,69 @@ public RawZone(JobArguments jobArguments,
...
         val startTime = System.currentTimeMillis();
 
-        String rowSource = table.getAs(SOURCE);
-        String rowTable = table.getAs(TABLE);
-        String rowOperation = table.getAs(OPERATION);
+        String rowSource = row.getAs(SOURCE);
+        String rowTable = row.getAs(TABLE);
 
-        logger.info("Processing batch with {} records for source: {} table: {}",
+        logger.info("Processing raw batch with {} records for source: {} table: {}",
```
Comment (carwynmoj) [inline src/main/java/uk/gov/justice/digital/zone/RawZone.java:79]: This is fine, but bear in mind that the class is also in the log output so we can see which zone it is already. I can show you some log examples if needed.

File: src/test/java/uk/gov/justice/digital/zone/RawZoneTest.java [hunk 4]
```diff
@@ -2,85 +2,250 @@
...
 import static org.mockito.Mockito.doNothing;
 import static org.mockito.Mockito.when;
 import static uk.gov.justice.digital.common.ResourcePath.createValidatedPath;
+import static uk.gov.justice.digital.converter.dms.DMS_3_4_6.Operation.*;
+import static uk.gov.justice.digital.converter.dms.DMS_3_4_6.ParsedDataFields.OPERATION;
 import static uk.gov.justice.digital.zone.Fixtures.*;
 
 @ExtendWith(MockitoExtension.class)
-class RawZoneTest {
+class RawZoneTest extends BaseSparkTest {
```
Comment (carwynmoj) [inline src/test/java/uk/gov/justice/digital/zone/RawZoneTest.java:35]: 👍 This makes sense and good to get some proper integration test coverage on this class.

**Review by carwynmoj — APPROVED** [review]
Looks good so far.

Added some observations. I reckon adding the cdc operations to the enum might be worth doing otherwise nothing else really needs changing.

**Review by dominic-messenger-moj — COMMENTED** [review]
Raw shouldn't filter, and all rows should be written as they come in (so shouldn't be sorted either).

File: src/main/java/uk/gov/justice/digital/zone/RawZone.java
[the remark this answers is not in the document]
Comment (koladeadewuyi-moj) [reply src/main/java/uk/gov/justice/digital/zone/RawZone.java:1]: The changes have now been made to have a single process for the RawZone.

**Pull request merged**